pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -236,11 +236,34 @@ func (az *Cloud) getSecurityRuleName(service *v1.Service, port v1.ServicePort, s
 		safePrefix := strings.Replace(sourceAddrPrefix, "/", "_", -1)
 		return fmt.Sprintf("shared-%s-%d-%s", port.Protocol, port.Port, safePrefix)
 	}
+	if sourceAddrPrefix == "" {
```

Comment:
sourceAddrPrefix is default to "Internet" by default, so why is this required?

Reply:
https://github.com/kubernetes/kubernetes/blob/ec4b3a231cfa19b8cfc5c5b8fbf7fac048914f87/pkg/cloudprovider/providers/azure/azure_loadbalancer.go#L1021-L1030

The logic for security rule name is changed such that if it's a shared rule, we append `Internet` or the sourceAddressPrefix, for example:`shared-PROTOCOL-PORT-Internet` and `shared-PROTOCOL-PORT-SOURCE-SourceRangePublicIP` respectively. If it's a not shared rule, then we do not append anything, for example: `UUID-PROTOCOL-PORT` since UUID is unique and we don't want to create a different rule name for each sourceAddressPrefix.

Reply:
could we keep it same as before? so that if the cluster is upgraded from old versions, old rules could still be recognized.

Reply:
If we keep the rule name the same as before, then we cannot collapse the rules into one as the name includes the source range as part of it. Do you have any suggestions?